File: Algorithms/1260. Shift 2D Grid/golang/solution.go
func shiftGrid(grid [][]int, k int) [][]int {
	numRows := len(grid)
	numCols := len(grid[0])
	var newGrid [][]int

	for row := 0; row < numRows; row++ {
		var newRow []int
		for col := 0; col < numCols; col++ {
			idx := row*numCols + col - k
			for idx < 0 {
				idx += numCols * numRows
			}

			newRow = append(newRow, grid[idx/numCols][idx%numCols])
		}
		newGrid = append(newGrid, newRow)
	}
	return newGrid
}

File: Algorithms/1260. Shift 2D Grid/javascript/solution.js
/**
 * @param {number[][]} grid
 * @param {number} k
 * @return {number[][]}
 */
var shiftGrid = function (grid, k) {
    const numRows = grid.length,
        numCols = grid[0].length;
    const newGrid = [];
    for (let i = 0; i < numRows; i++) {
        newGrid[i] = new Array(numCols).fill(0);
    }

    for (let row = 0; row < numRows; row++) {
        for (let col = 0; col < numCols; col++) {
            const newCol = (col + k) % numCols;
            const newRow = (row + Math.floor((col + k) / numCols)) % numRows;
            newGrid[newRow][newCol] = grid[row][col];
        }
    }

    return newGrid;
};
